dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -221,249 +232,412 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
         }
     }
 
-    void PushNotificationManager::RegisterActivator(PushNotificationActivationInfo const& details)
+    bool PushNotificationManager::IsBackgroundTaskRegistered(winrt::hstring const& backgroundTaskFullName)
     {
-        THROW_HR_IF(E_NOTIMPL, !::Microsoft::Windows::PushNotifications::Feature_PushNotifications::IsEnabled());
-
-        try
-        {
-            THROW_HR_IF_NULL(E_INVALIDARG, details);
-
-            auto registrationActivators{ details.Activators() };
+        auto tasks{ BackgroundTaskRegistration::AllTasks() };
+        bool isTaskRegistered{ std::any_of(std::begin(tasks), std::end(tasks),
+            [&](auto&& task)
+            {
+                auto name{ task.Value().Name() };
 
-            auto isBackgroundTaskFlagSet{ WI_IsAnyFlagSet(registrationActivators, PushNotificationRegistrationActivators::PushTrigger | PushNotificationRegistrationActivators::ComActivator) };
-            THROW_HR_IF(E_INVALIDARG, isBackgroundTaskFlagSet && !IsActivatorSupported(registrationActivators));
+                return name == backgroundTaskFullName;
+            }) };
 
-            auto isProtocolActivatorSet{ WI_IsFlagSet(registrationActivators, PushNotificationRegistrationActivators::ProtocolActivator) };
-            THROW_HR_IF(E_INVALIDARG, isProtocolActivatorSet && !IsActivatorSupported(registrationActivators));
+        return isTaskRegistered;
+    }
 
-            if (isProtocolActivatorSet)
+    IBackgroundTaskRegistration RetreiveBackgroundTaskRegistration(winrt::hstring const& backgroundTaskFullName)
+    {
+        auto tasks{ BackgroundTaskRegistration::AllTasks() };
+        IBackgroundTaskRegistration registration{};
+
+        // Custom version of std::find since we are looking for the name
+        // of the task not the registration.
+        auto first{ std::begin(tasks) };
+        auto last{ std::end(tasks) };
+        while (first != last)
```

Comment:
So there is no expression you can use to achieve the same without introducing a loop?

Reply:
Replacing with std::find_if